### ISSUE TITLE
querying for a specific mutation returns also genomes that do not contain this mutation

### ISSUE THREAD

**Issue by akifoss** [rki-mf1/covsonar#78]
When using e.g.,

`
sonar.py match -db db -i S:R346X --lineage BA.5.3 --with-sublineage`

among the returned genomes there are some which do not contain the R346X mutation.

In the above example, there were only n=109 genomes containing the R346X mutation in a total of n=867 returned genomes.

**Comment by stephan-fuchs** [comment]
This is actually a feature and not a bug.
X is the IUPAC code for any amino acid. CovSonar interprets X in query profiles accordingly as all non-reference alleles at the corresponding protein site. This explains the behavior you describe. Does this answer your question?

**Comment by akifoss** [comment]
In the above example of querying for BA.5.3 genomes with S:R346X, the majority of the returned genomes do not contain any mutation at position 346. The ambiguity is seen in the n=109 genomes (R changes to T or S or I), but there is no amino acid change at position 346 for the rest of the n=867 genomes.

**Comment by stephan-fuchs** [comment]
Sorry, then I got you wrong. I will have a look on this!

**Comment by stephan-fuchs** [comment]
I apologize for the delay in my response!

To address your concern, can you please confirm that you are using the latest version of covSonar, v1.1.7? If not, please install it using the following commands:

```bash
git clone -b v1.1.7 https://github.com/rki-mf1/covsonar.git
cd covsonar
sonar.py --version
```

Additionally, please note that if you wish to include any ambiguous mutations in the returned mutation profiles, you must use the "--ambig" option.

. Please let me know if the problem persists or you have any further questions or issues.

**Comment by akifoss** [comment]
thanks for the response!

This is indeed covSonar v1.1.7 which I am using. 

The ambiguous mutations can be successfully retrieved, the issue is not there. 
But when querying for an ambiguous mutation, there are hits returned that do not contain the mutated site at all. In the above example, there are many hits without any mutation at position S:R346.

**Comment by matthuska** [comment]
Hey @akifoss , I took a few minutes to dig into this a bit. Basically what @stephan-fuchs said about `--ambig` is right.

You're querying for "S:R346X", so any mutation at position S:R346. Some of the sequences in your result set are only matching because they contain ambiguous amino acids at position S:R346, but because you probably didn't use the `--ambig` flag, then these positions are not included in the aa_profile field, so it seems like hits are being returned that don't have that mutation at all. Do the exact same query, with the `--ambig` flag included, and you should see a bunch of S:R346X mutations in your amino acid profiles that previously looked like they didn't have any mutation at S:R346.

In short, both:
```
sonar.py match --db mydb.db -i S:R346X
```
and
```
sonar.py match --db mydb.db -i S:R346X --ambig
```
should return the exact same sequences, but the first command might have sequences where the `aa_profile` field is completely lacking anything at `S:R346`. The second query should always include something at `S:R346`, including `S:R346X` which would be missing in the first case. The `--ambig` flag only impacts the **output**, not the match query itself.

I'll close this bug for now but feel free to reopen it if the answer I gave doesn't seem correct.
